agb/src/interrupt.rs

```diff
@@ -21,37 +28,344 @@ pub enum Interrupt {
 const ENABLED_INTERRUPTS: MemoryMapped<u16> = unsafe { MemoryMapped::new(0x04000200) };
 const INTERRUPTS_ENABLED: MemoryMapped<u16> = unsafe { MemoryMapped::new(0x04000208) };
 
-pub fn enable(interrupt: Interrupt) {
+fn enable(interrupt: Interrupt) {
     let _interrupt_token = temporary_interrupt_disable();
+    other_things_to_enable_interrupt(interrupt);
     let interrupt = interrupt as usize;
     let enabled = ENABLED_INTERRUPTS.get() | (1 << (interrupt as u16));
     ENABLED_INTERRUPTS.set(enabled);
 }
 
-pub fn disable(interrupt: Interrupt) {
+fn disable(interrupt: Interrupt) {
     let _interrupt_token = temporary_interrupt_disable();
+    other_things_to_disable_interrupt(interrupt);
     let interrupt = interrupt as usize;
     let enabled = ENABLED_INTERRUPTS.get() & !(1 << (interrupt as u16));
     ENABLED_INTERRUPTS.set(enabled);
 }
 
-pub struct Disable {}
+fn other_things_to_enable_interrupt(interrupt: Interrupt) {
+    match interrupt {
+        Interrupt::VBlank => {
+            DISPLAY_STATUS.set_bits(1, 1, 3);
+        }
+        Interrupt::HBlank => {
+            DISPLAY_STATUS.set_bits(1, 1, 4);
+        }
+        _ => {}
+    }
+}
+
+fn other_things_to_disable_interrupt(interrupt: Interrupt) {
+    match interrupt {
+        Interrupt::VBlank => {
+            DISPLAY_STATUS.set_bits(0, 1, 3);
+        }
+        Interrupt::HBlank => {
+            DISPLAY_STATUS.set_bits(0, 1, 4);
+        }
+        _ => {}
+    }
+}
+
+struct Disable {}
 
 impl Drop for Disable {
     fn drop(&mut self) {
         enable_interrupts();
     }
 }
 
-pub fn temporary_interrupt_disable() -> Disable {
+fn temporary_interrupt_disable() -> Disable {
     disable_interrupts();
     Disable {}
 }
 
-pub fn enable_interrupts() {
+fn enable_interrupts() {
     INTERRUPTS_ENABLED.set(1);
 }
 
-pub(crate) fn disable_interrupts() {
+fn disable_interrupts() {
```

Comment:
feels like this could be extracted into its own file with the `Interrupt` enum above to keep this a bit more self-contained and single purpose?